adapters/src/transport/file.rs

```diff
@@ -0,0 +1,331 @@
+use super::{InputConsumer, InputEndpoint, InputTransport};
+use crate::PipelineState;
+use anyhow::{Error as AnyError, Result as AnyResult};
+use crossbeam::sync::{Parker, Unparker};
+use num_traits::FromPrimitive;
+use serde::Deserialize;
+use serde_yaml::Value as YamlValue;
+use std::{
+    borrow::Cow,
+    fs::File,
+    io::{BufRead, BufReader},
+    sync::{
+        atomic::{AtomicU32, Ordering},
+        Arc,
+    },
+    thread::{sleep, spawn},
+    time::Duration,
+};
+
+const SLEEP_MS: u64 = 200;
+
+/// `InputTransport` implementation that reads data from file.
+pub struct FileInputTransport;
+
+impl InputTransport for FileInputTransport {
+    fn name(&self) -> Cow<'static, str> {
+        Cow::Borrowed("file")
+    }
+
+    fn new_endpoint(
+        &self,
+        config: &YamlValue,
+        consumer: Box<dyn InputConsumer>,
+    ) -> AnyResult<Box<dyn InputEndpoint>> {
+        let config = FileInputConfig::deserialize(config)?;
+        let mut ep = FileInputEndpoint::new(config);
+        ep.connect(consumer)?;
+        Ok(Box::new(ep))
+    }
+}
+
+#[derive(Deserialize)]
+struct FileInputConfig {
+    /// File path.
+    path: String,
+
+    /// Read buffer size.
+    ///
+    /// Default: when this parameter is not specified, a platform-specific
+    /// default is used.
+    buffer_size: Option<usize>,
+
+    /// Enable file following.
+    ///
+    /// When `false`, the endpoint outputs an [`eof`](`InputConsumer::eof`)
```

Comment:
I guess the answer is "no" to the above question.
But I wonder what happens if the file is for example 'truncated'.